### PR TITLE
conditional imports

### PR DESCRIPTION
**Pull request by hwchase17**

### CODE REVIEW

File: libs/langchain/langchain/__init__.py [hunk 1]
```diff
@@ -1,57 +1,12 @@
...
-from langchain.utilities.google_search import GoogleSearchAPIWrapper
-from langchain.utilities.google_serper import GoogleSerperAPIWrapper
-from langchain.utilities.powerbi import PowerBIDataset
-from langchain.utilities.searx_search import SearxSearchWrapper
-from langchain.utilities.serpapi import SerpAPIWrapper
-from langchain.utilities.sql_database import SQLDatabase
-from langchain.utilities.wikipedia import WikipediaAPIWrapper
-from langchain.utilities.wolfram_alpha import WolframAlphaAPIWrapper
-from langchain.vectorstores import FAISS, ElasticVectorSearch
+from typing import TYPE_CHECKING, Any, Optional
```
Comment (nfcampos) [inline libs/langchain/langchain/__init__.py:5]: Should these any and optional be inside the type checking branch too?

File: libs/langchain/langchain/__init__.py [hunk 2]
```diff
@@ -62,10 +17,198 @@
 
 verbose: bool = False
 debug: bool = False
-llm_cache: Optional[BaseCache] = None
+llm_cache: Optional["BaseCache"] = None
+
+
+def __getattr__(name: str) -> Any:
+    warnings.warn(f"Importing {name} is no longer supported.")
```
Comment (nfcampos) [inline libs/langchain/langchain/__init__.py:24]: This should be reworded to something like "importing x from langchain root module is no longer supported, it should now be imported from langchain.x see more details at http://"

**Review by nfcampos — APPROVED** [review]

File: libs/langchain/langchain/__init__.py (hunk 1 above)
Comment (hwchase17) [reply libs/langchain/langchain/__init__.py:5]: eh that cant really matter right?

File: libs/langchain/langchain/__init__.py (hunk 2 above)
Comment (hwchase17) [reply libs/langchain/langchain/__init__.py:24]: what should i link to?

**Review by baskaryan — COMMENTED** [review]
worth adding unit test?

**Pull request merged**